lib/sqlalchemy/dialects/postgresql/base.py

```diff
@@ -3049,15 +3049,18 @@ def initialize(self, connection):
         super().initialize(connection)
 
         # https://www.postgresql.org/docs/9.3/static/release-9-2.html#AEN116689
-        self.supports_smallserial = self.server_version_info >= (9, 2)
+        self.supports_smallserial = self.server_version_info and self.server_version_info >= (9, 2)
 
-        self._set_backslash_escapes(connection)
+        try: 
+            self._set_backslash_escapes(connection)
+        except:
```

Comment:
this would cause silent errors

Reply:
Agree., also my concern when I hacked it to hear from you. 

A database should better comply but I'm curious whether sqlalchemy has a fallback solution when a database for whatever reason doesn't support this. Is there a legacy solution in history or do we know for sure where needs to be retooled for people to support that dialect without the trials and errors?

Reply:
the "this" here is custom code that's written to support postgresql.

If a database does need to do something different, the dialect can decide to handle it in a different way.